client/src/real-estates-list/RealEstateListPage.js

```diff
@@ -0,0 +1,28 @@
+import RealEstatePreview from "./RealEstatePreview";
+import { useRealEstates } from "../real-estates";
+
+export default function RealEstateListPage() {
+
+  const realEstates = useRealEstates();
+
+  const RealEstatesOnSale = () => (
+    realEstates.map((realEstate) => {
+      if(realEstate.onSale) {
+        return (<RealEstatePreview realEstate={realEstate} key={realEstate.tokenId}/>)
+    }
+  })
```

Comment:
Ton soucis de `key must be unique` c'est dû au fait que ton map renvoie soit un élément JSX, soit `undefined`, et `undefined` n'a pas de `key`. Il faudrait plutôt d'abord `.filter()` ET ensuite `.map()`.
Deuxième chose, dans ce cas là c'est mieux de compute directement un JSX plutôt que de créer un component. En d'autres termes, c'est mieux que le type de ta variable soit `ReactNode` que `React.Component`. Enfin, pour un cas simple comme ça, tu peux directement l'inline dans le rendu du component (voir mon commentaire en dessous pour le résultat final)